### ISSUE TITLE
Allow setting settings by syntax

### ISSUE THREAD

**Issue by spencerlyon2** [dvcrn/proton#38]
In my config I have things like:

```
".coffee.source":
  editor:
    tabLength: 2
".bibtex.text":
  editor:
    tabLength: 2
".julia.source":
  editor:
    tabLength: 2
  "multi-wrap-guide":
    columns: [
      72
      80
      92
    ]
```

If I try to add something like `[".julia.source.editor.tabLength" 2]` in my `~/.proton` I get 

```
  "":
    julia:
      source:
        editor:
          tabLength: 2
```

I think we just need to special case a leading `.` in the config string when parsing configs

**Comment by spencerlyon2** [comment]
Although it is a bit tricky as we will need to think about where to start applying normal behavior to `.`

I _think_ that we will always see syntax specific settings of the form `.(text|source).scopename.SETTINGS`

**Comment by dvcrn** [comment]
So what proton currently does is, it just takes the config string (`".latex.tex.text") and passes that straight into the atom API for setting a setting. 

Yeah we should definitely try to collect all the settings for the user and push it into an array in these cases or something like that

**Comment by spencerlyon2** [comment]
Ping, I'm still missing this. Has anyone put any thought into how this might work.

What if we just have a "raw-config" variable people can set in their `.proton` that we copy/paste directly into `~/.atom/config.cson`. It would be totally un-checked and un-verified and would escape the rest of the proton system, but it would give users access to set anything atom can handle.

**Comment by geksilla** [comment]
@spencerlyon2 for now you can disable wiping configuration and put configs manually via Settings View :unamused: I will try to take care about this issue when will have some time.

**Comment by dvcrn** [comment]
I think that raw-config thing can be a good short time solution. I'm just worried that people will end up using it for everything 😅

**Comment by geksilla** [comment]
Fixed in #169
